### PR TITLE
add ssh option for all servers linux and windows

### PR DESCRIPTION
**Pull request by jackyalbo**

### CODE REVIEW

File: lib/pkgcloud/azure-v2/compute/client/servers.js [hunk 1]
```diff
@@ -40,19 +41,17 @@ function getLimits(callback) {
 function getServers(callback) {
   var self = this;
 
-  self.login(function (err) {
+  self.login(err => {
```
Comment (morsh) [inline lib/pkgcloud/azure-v2/compute/client/servers.js:44]: This format is not supported in the "declared" version of this repo (node 0.10 or something) so it has to be function (err) {...}

[Here and everywhere]

File: lib/pkgcloud/azure-v2/compute/client/servers.js [hunk 2]
```diff
@@ -143,71 +149,110 @@ function createServer(options, callback) {
...
-        template.resources[vmIndex].properties.storageProfile.dataDisks.push({
-          'name': 'datadisk' + idx.toString(),
-          'diskSizeGB': '100',
-          'lun': 0,
-          'vhd': {
-            'uri': '[concat(reference(concat(\'Microsoft.Storage/storageAccounts/\', variables(\'storageAccountName\')), \'2016-01-01\').primaryEndpoints.blob, parameters(\'storageContainerName\'),\'/\', \'' + ddName + '\', \'.vhd\')]'
-          },
-          'createOption': 'Empty'
-        });
-      });
```
Comment (morsh) [inline lib/pkgcloud/azure-v2/compute/client/servers.js:174]: Why did you remove those lines?
Those lines enable adding multiple data disks when creating/updating a vm

File: lib/pkgcloud/azure-v2/compute/client/servers.js [hunk 3]
```diff
@@ -143,71 +149,110 @@ function createServer(options, callback) {
...
-          'publisher': 'Microsoft.Compute',
-          'type': 'CustomScriptExtension',
-          'typeHandlerVersion': '1.4',
-          'settings': {
-            'fileUris': [
-              'https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-vm-winrm-windows/ConfigureWinRM.ps1',
-              'https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-vm-winrm-windows/makecert.exe',
-              'https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-vm-winrm-windows/winrmconf.cmd'
-            ],
-            'commandToExecute': '[concat(\'powershell -ExecutionPolicy Unrestricted -file ConfigureWinRM.ps1 \',variables(\'hostDNSNameScriptArgument\'))]'
```
Comment (morsh) [inline lib/pkgcloud/azure-v2/compute/client/servers.js:197]: Also not clear - for windows you need an extension to enable remote connect - why not add this as part of the ARM deployment and create a separate deployment?

File: lib/pkgcloud/azure-v2/compute/client/servers.js [hunk 4]
```diff
@@ -143,71 +149,110 @@ function createServer(options, callback) {
...
+            "publisher": "Microsoft.Compute",
+            "typeHandlerVersion": "1.8",
+            "type": "CustomScriptExtension",
+            "settings": {
+              "fileUris": ["https://pluginsstorage.blob.core.windows.net/agentscripts/ssh.ps1"],
+              "commandToExecute": `powershell -File ssh.ps1 ${hostname}\\${options.username} ${options.password}`
+            },
+            "protectedSettings": {
+              "storageAccountName": 'pluginsstorage',
+              "storageAccountKey": 'bHabDjY34dXwITjXEasmQxI84QinJqiBZHiU+Vc1dqLNSKQxvFrZbVsfDshPriIB+XIaFVaQ2R3ua1YMDYYfHw=='
```
Comment (morsh) [inline lib/pkgcloud/azure-v2/compute/client/servers.js:212]: Not sure what this is for, but you can't add constant personal information like:
https://pluginsstorage.blob.core.windows.net/agentscripts/ssh.ps1
pluginsstorage
bHabDjY34dXwITjXEasmQxI84QinJqiBZHiU+Vc1dqLNSKQxvFrZbVsfDshPriIB+XIaFVaQ2R3ua1YMDYYfHw==

[Same for Linux]

**Review by morsh — COMMENTED** [review]
@jackyalbo - I wrote some comments on the changes + it's a little hard to review since those changes include code structure changes, so in many of the changes I'm not sure if they're structure or logic.

File: lib/pkgcloud/azure-v2/compute/client/servers.js (hunk 1 above)
Comment (jackyalbo) [reply lib/pkgcloud/azure-v2/compute/client/servers.js:44]: ok, ill fix

File: lib/pkgcloud/azure-v2/compute/client/servers.js (hunk 2 above)
Comment (jackyalbo) [reply lib/pkgcloud/azure-v2/compute/client/servers.js:174]: it was for testing - but i don't want data disk by default? can we make it optional?

File: lib/pkgcloud/azure-v2/compute/client/servers.js (hunk 3 above)
Comment (jackyalbo) [reply lib/pkgcloud/azure-v2/compute/client/servers.js:197]: so the thing is i want to use ssh instead of winRM... I changed this lines with ones that adds ssh support.

File: lib/pkgcloud/azure-v2/compute/client/servers.js (hunk 4 above)
Comment (jackyalbo) [reply lib/pkgcloud/azure-v2/compute/client/servers.js:212]: you are correct - my bad - I'll need to find public repo to use instead...

File: lib/pkgcloud/azure-v2/compute/client/servers.js (hunk 2 above)
Comment (morsh) [reply lib/pkgcloud/azure-v2/compute/client/servers.js:174]: they are optional:
if (options.storageDataDiskNames > 0) ...

File: lib/pkgcloud/azure-v2/compute/client/servers.js (hunk 3 above)
Comment (morsh) [reply lib/pkgcloud/azure-v2/compute/client/servers.js:197]: Right, sounds great, but this means two things:

1. Would be better to add this using `template.resources[vmIndex].resources = [{...`
2. For this to work E2E you'll need to change cloud-cd to only support ssh (for windows/linux) - but this is not in the scope of this repo
